packages/app-store/googlecalendar/lib/CalendarService.ts

```diff
@@ -85,13 +85,15 @@ export default class GoogleCalendarService implements Calendar {
     this.credential = credential;
     this.auth = this.initGoogleAuth(credential);
     this.log = log.getSubLogger({ prefix: [`[[lib] ${this.integrationName}`] });
-    this.credential = credential;
   }
 
   private async getMyGoogleAuthSingleton() {
-    const googleCredentials = OAuth2UniversalSchema.parse(this.credential.key);
+    if (this.myGoogleAuth) {
+      return this.myGoogleAuth;
+    }
     const { client_id, client_secret, redirect_uris } = await getGoogleAppKeys();
-    this.myGoogleAuth = this.myGoogleAuth || new MyGoogleAuth(client_id, client_secret, redirect_uris[0]);
+    const googleCredentials = OAuth2UniversalSchema.parse(this.credential.key);
+    this.myGoogleAuth = new MyGoogleAuth(client_id, client_secret, redirect_uris[0]);
     this.myGoogleAuth.setCredentials(googleCredentials);
```

Comment:
Now `setCredentials` is also called only the first time. As `getMyGoogleAuthSingleton` can be called multiple times, we don't need to set it again and again.